Error during compilation for musl libc in Empire/Diplomacy.cpp

Bug Report
==========

Environment
-----------
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example a crashing hosted server may not need
information about the graphic card, but maybe depend on the operating system you
use.  When in doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**: 0.5.0.1
* **Operating System**: Void Linux
* **Graphic card used**: N/A
* **Fetched as** <!-- delete lines below that don't apply -->
  * Compiled from source


Description
-----------
I'm trying to update package in Void Linux repository for Freeorion.
I have some problems with cross compilation for musl libc.
I'm receiving error when compiling first target, `Empire/Diplomacy.cpp.o`.
I tried to downgrade for 0.5, but the error persisted.
I have no idea why it's failing. Wrong GCC version? I'm using 13.2.0
For me it looks that Void Linux updated some library or something and thus now Freeorion compilation fails, but maybe I'm wrong.

If you need more informations I'll be happy to provide them :)

Expected Result
---------------
Freeorion is compiled



Some failed CI jobs with logs:
https://github.com/void-linux/void-packages/actions/runs/8620450624

I'm pasting here my local log if github's log is lost
<details>
  <summary>Log</summary>
  
```
[1/269] Building CXX object CMakeFiles/freeorioncommon.dir/Empire/Diplomacy.cpp.o
FAILED: CMakeFiles/freeorioncommon.dir/Empire/Diplomacy.cpp.o 
/usr/bin/ccache /builddir/.xbps-freeorion/wrappers/x86_64-linux-musl-c++ -DBINPATH=\"/usr/bin\" -DBOOST_ALL_DYN_LINK -DBOOST_PHOENIX_STL_TUPLE_H_ -DENABLE_BINRELOC -DFREEORION_BUILD_COMMON -DFREEORION_LINUX -DSHAREPATH=\"/usr/share\" -Dfreeorioncommon_EXPORTS -I /usr/x86_64-linux-musl/usr/include/python3.12 -DNDEBUG -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -mtune=generic   -I/usr/x86_64-linux-musl/usr/include -ffile-prefix-map=/builddir/freeorion-0.5.0.1/build=. -std=gnu++17 -fPIC -fvisibility=hidden -Wall -MD -MT CMakeFiles/freeorioncommon.dir/Empire/Diplomacy.cpp.o -MF CMakeFiles/freeorioncommon.dir/Empire/Diplomacy.cpp.o.d -o CMakeFiles/freeorioncommon.dir/Empire/Diplomacy.cpp.o -c /builddir/freeorion-0.5.0.1/Empire/Diplomacy.cpp
In file included from ../Empire/Diplomacy.h:6,
                 from ../Empire/Diplomacy.cpp:1:
../Empire/../util/Enum.h:33:6: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
   33 | enum class \
      | ~~~~ ^~~~~
../Empire/../util/Enum.h:241:5: note: in expansion of macro 'FO_DEF_ENUM'
  241 |     FO_DEF_ENUM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:35:1: error: found ':' in nested-name-specifier, expected '::'
   35 | : int8_t { \
      | ^
../Empire/../util/Enum.h:241:5: note: in expansion of macro 'FO_DEF_ENUM'
  241 |     FO_DEF_ENUM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
In file included from /usr/x86_64-linux-musl/usr/include/boost/preprocessor/comparison/not_equal.hpp:821,
                 from /usr/x86_64-linux-musl/usr/include/boost/preprocessor/comparison/equal.hpp:17,
                 from ../Empire/../util/Enum.h:9:
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:34:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   34 | FO_ENUM_NAME_FROM_TYPENAME(typeName) \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:241:5: note: in expansion of macro 'FO_DEF_ENUM'
  241 |     FO_DEF_ENUM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:35:10: error: expected unqualified-id before '{' token
   35 | : int8_t { \
      |          ^
../Empire/../util/Enum.h:241:5: note: in expansion of macro 'FO_DEF_ENUM'
  241 |     FO_DEF_ENUM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' was not declared in this scope
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:52:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   52 | FO_ENUM_NAME_FROM_TYPENAME(typeName) \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:242:5: note: in expansion of macro 'FO_DEF_ENUM_TOSTRING'
  242 |     FO_DEF_ENUM_TOSTRING(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:79:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   79 | FO_ENUM_NAME_FROM_TYPENAME(typeName) value) { \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:243:5: note: in expansion of macro 'FO_DEF_ENUM_OSTREAM'
  243 |     FO_DEF_ENUM_OSTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h: In function 'std::ostream& operator<<(std::ostream&, int)':
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:68:10: note: in definition of macro 'FO_DEF_ENUM_OSTREAM_CASE'
   68 |     case data::BOOST_PP_TUPLE_ELEM(0, elem): \
      |          ^~~~
../Empire/../util/Enum.h:82:13: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   82 |             FO_ENUM_NAME_FROM_TYPENAME(typeName), \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:243:5: note: in expansion of macro 'FO_DEF_ENUM_OSTREAM'
  243 |     FO_DEF_ENUM_OSTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:68:10: note: in definition of macro 'FO_DEF_ENUM_OSTREAM_CASE'
   68 |     case data::BOOST_PP_TUPLE_ELEM(0, elem): \
      |          ^~~~
../Empire/../util/Enum.h:82:13: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   82 |             FO_ENUM_NAME_FROM_TYPENAME(typeName), \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:243:5: note: in expansion of macro 'FO_DEF_ENUM_OSTREAM'
  243 |     FO_DEF_ENUM_OSTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:68:10: note: in definition of macro 'FO_DEF_ENUM_OSTREAM_CASE'
   68 |     case data::BOOST_PP_TUPLE_ELEM(0, elem): \
      |          ^~~~
../Empire/../util/Enum.h:82:13: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   82 |             FO_ENUM_NAME_FROM_TYPENAME(typeName), \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:243:5: note: in expansion of macro 'FO_DEF_ENUM_OSTREAM'
  243 |     FO_DEF_ENUM_OSTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:68:10: note: in definition of macro 'FO_DEF_ENUM_OSTREAM_CASE'
   68 |     case data::BOOST_PP_TUPLE_ELEM(0, elem): \
      |          ^~~~
../Empire/../util/Enum.h:82:13: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   82 |             FO_ENUM_NAME_FROM_TYPENAME(typeName), \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:243:5: note: in expansion of macro 'FO_DEF_ENUM_OSTREAM'
  243 |     FO_DEF_ENUM_OSTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:68:10: note: in definition of macro 'FO_DEF_ENUM_OSTREAM_CASE'
   68 |     case data::BOOST_PP_TUPLE_ELEM(0, elem): \
      |          ^~~~
../Empire/../util/Enum.h:82:13: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
   82 |             FO_ENUM_NAME_FROM_TYPENAME(typeName), \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:243:5: note: in expansion of macro 'FO_DEF_ENUM_OSTREAM'
  243 |     FO_DEF_ENUM_OSTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h: At global scope:
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:105:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  105 | FO_ENUM_NAME_FROM_TYPENAME(typeName)& value) \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:244:5: note: in expansion of macro 'FO_DEF_ENUM_ISTREAM'
  244 |     FO_DEF_ENUM_ISTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h: In function 'std::istream& operator>>(std::istream&, int&)':
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:96:17: note: in definition of macro 'FO_DEF_ENUM_ISTREAM_CASE'
   96 |         value = data::BOOST_PP_TUPLE_ELEM(0, elem);
      |                 ^~~~
../Empire/../util/Enum.h:244:5: note: in expansion of macro 'FO_DEF_ENUM_ISTREAM'
  244 |     FO_DEF_ENUM_ISTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:96:17: note: in definition of macro 'FO_DEF_ENUM_ISTREAM_CASE'
   96 |         value = data::BOOST_PP_TUPLE_ELEM(0, elem);
      |                 ^~~~
../Empire/../util/Enum.h:244:5: note: in expansion of macro 'FO_DEF_ENUM_ISTREAM'
  244 |     FO_DEF_ENUM_ISTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:96:17: note: in definition of macro 'FO_DEF_ENUM_ISTREAM_CASE'
   96 |         value = data::BOOST_PP_TUPLE_ELEM(0, elem);
      |                 ^~~~
../Empire/../util/Enum.h:244:5: note: in expansion of macro 'FO_DEF_ENUM_ISTREAM'
  244 |     FO_DEF_ENUM_ISTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:96:17: note: in definition of macro 'FO_DEF_ENUM_ISTREAM_CASE'
   96 |         value = data::BOOST_PP_TUPLE_ELEM(0, elem);
      |                 ^~~~
../Empire/../util/Enum.h:244:5: note: in expansion of macro 'FO_DEF_ENUM_ISTREAM'
  244 |     FO_DEF_ENUM_ISTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' has not been declared
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:96:17: note: in definition of macro 'FO_DEF_ENUM_ISTREAM_CASE'
   96 |         value = data::BOOST_PP_TUPLE_ELEM(0, elem);
      |                 ^~~~
../Empire/../util/Enum.h:244:5: note: in expansion of macro 'FO_DEF_ENUM_ISTREAM'
  244 |     FO_DEF_ENUM_ISTREAM(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h: At global scope:
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' was not declared in this scope
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:134:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  134 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:134:44: error: template argument 1 is invalid
  134 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      |                                            ^~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:134:55: error: template argument 1 is invalid
  134 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      |                                                       ^~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:134:55: error: template argument 2 is invalid
  134 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      |                                                       ^~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h: In function 'int DiplomaticStatusValues()':
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' was not declared in this scope; did you mean 'DiplomaticStatusValues'?
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:137:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  137 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> ret{ \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:137:44: error: template argument 1 is invalid
  137 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> ret{ \
      |                                            ^~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:137:55: error: template argument 1 is invalid
  137 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> ret{ \
      |                                                       ^~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:137:55: error: template argument 2 is invalid
  137 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> ret{ \
      |                                                       ^~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' is not a class, namespace, or enumeration
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:125:6: note: in definition of macro 'FO_DEF_ENUM_ITERATE_VALUE'
  125 |     {data::BOOST_PP_TUPLE_ELEM(0, elem), BOOST_PP_STRINGIZE(BOOST_PP_TUPLE_ELEM(0, elem)) },
      |      ^~~~
../Empire/../util/Enum.h:139:9: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  139 |         FO_ENUM_NAME_FROM_TYPENAME(typeName), values) \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' is not a class, namespace, or enumeration
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:125:6: note: in definition of macro 'FO_DEF_ENUM_ITERATE_VALUE'
  125 |     {data::BOOST_PP_TUPLE_ELEM(0, elem), BOOST_PP_STRINGIZE(BOOST_PP_TUPLE_ELEM(0, elem)) },
      |      ^~~~
../Empire/../util/Enum.h:139:9: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  139 |         FO_ENUM_NAME_FROM_TYPENAME(typeName), values) \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' is not a class, namespace, or enumeration
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:125:6: note: in definition of macro 'FO_DEF_ENUM_ITERATE_VALUE'
  125 |     {data::BOOST_PP_TUPLE_ELEM(0, elem), BOOST_PP_STRINGIZE(BOOST_PP_TUPLE_ELEM(0, elem)) },
      |      ^~~~
../Empire/../util/Enum.h:139:9: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  139 |         FO_ENUM_NAME_FROM_TYPENAME(typeName), values) \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' is not a class, namespace, or enumeration
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:125:6: note: in definition of macro 'FO_DEF_ENUM_ITERATE_VALUE'
  125 |     {data::BOOST_PP_TUPLE_ELEM(0, elem), BOOST_PP_STRINGIZE(BOOST_PP_TUPLE_ELEM(0, elem)) },
      |      ^~~~
../Empire/../util/Enum.h:139:9: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  139 |         FO_ENUM_NAME_FROM_TYPENAME(typeName), values) \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' is not a class, namespace, or enumeration
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:125:6: note: in definition of macro 'FO_DEF_ENUM_ITERATE_VALUE'
  125 |     {data::BOOST_PP_TUPLE_ELEM(0, elem), BOOST_PP_STRINGIZE(BOOST_PP_TUPLE_ELEM(0, elem)) },
      |      ^~~~
../Empire/../util/Enum.h:139:9: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  139 |         FO_ENUM_NAME_FROM_TYPENAME(typeName), values) \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:137:58: error: scalar object 'ret' requires one element in initializer
  137 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> ret{ \
      |                                                          ^~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h: At global scope:
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' was not declared in this scope; did you mean 'DiplomaticStatusValues'?
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:148:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  148 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:148:44: error: template argument 1 is invalid
  148 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      |                                            ^~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:148:55: error: template argument 1 is invalid
  148 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      |                                                       ^~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:148:55: error: template argument 2 is invalid
  148 | FO_ENUM_NAME_FROM_TYPENAME(typeName), std::string_view>> \
      |                                                       ^~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' was not declared in this scope; did you mean 'DiplomaticStatusValues'?
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:150:1: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  150 | FO_ENUM_NAME_FROM_TYPENAME(typeName) >){ return \
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Enum.h:150:38: error: template argument 1 is invalid
  150 | FO_ENUM_NAME_FROM_TYPENAME(typeName) >){ return \
      |                                      ^
../Empire/../util/Enum.h:245:5: note: in expansion of macro 'FO_DEF_ENUM_ITERATE'
  245 |     FO_DEF_ENUM_ITERATE(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:13:6: error: 'DiplomaticStatus' does not name a type
   13 |     (DiplomaticStatus),
      |      ^~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:162:8: note: in expansion of macro 'FO_ENUM_NAME_FROM_TYPENAME'
  162 | inline FO_ENUM_NAME_FROM_TYPENAME(typeName) \
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Enum.h:246:5: note: in expansion of macro 'FO_DEF_ENUM_FROM_STRING'
  246 |     FO_DEF_ENUM_FROM_STRING(typeName)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/Diplomacy.h:24:10: warning: elaborated-type-specifier for a scoped enum must not use the 'class' keyword
   24 |     enum class Type : int8_t {
      |     ~~~~ ^~~~~
      |          -----
../Empire/Diplomacy.h:24:16: error: use of enum 'Type' without previous declaration
   24 |     enum class Type : int8_t {
      |                ^~~~
../Empire/Diplomacy.h:24:23: error: 'int8_t' was not declared in this scope
   24 |     enum class Type : int8_t {
      |                       ^~~~~~
../Empire/Diplomacy.h:9:1: note: 'int8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
    8 | #include "../universe/ConstantsFwd.h"
  +++ |+#include <cstdint>
    9 | 
../Empire/Diplomacy.h:24:30: error: default member initializer for unnamed bit-field
   24 |     enum class Type : int8_t {
      |                              ^
../Empire/Diplomacy.h:37:70: error: 'Type' has not been declared
   37 |     DiplomaticMessage(int sender_empire_id, int recipient_empire_id, Type type);
      |                                                                      ^~~~
../Empire/Diplomacy.h:49:5: error: 'Type' does not name a type; did you mean 'typeof'?
   49 |     Type m_type = Type::INVALID;
      |     ^~~~
      |     typeof
../Empire/Diplomacy.h: In member function 'auto DiplomaticMessage::GetType() const':
../Empire/Diplomacy.h:39:58: error: 'm_type' was not declared in this scope; did you mean 'wctype'?
   39 |     [[nodiscard]] auto GetType() const noexcept { return m_type; }
      |                                                          ^~~~~~
      |                                                          wctype
../Empire/Diplomacy.h: At global scope:
../Empire/Diplomacy.h:58:66: error: 'DiplomaticStatus' has not been declared
   58 |     DiplomaticStatusUpdateInfo(int empire1_id_, int empire2_id_, DiplomaticStatus status);
      |                                                                  ^~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:61:5: error: 'DiplomaticStatus' does not name a type
   61 |     DiplomaticStatus    diplo_status = DiplomaticStatus::INVALID_DIPLOMATIC_STATUS;
      |     ^~~~~~~~~~~~~~~~
In file included from ../Empire/../util/OptionsDB.h:5,
                 from ../Empire/../util/GameRules.h:4,
                 from ../Empire/Diplomacy.cpp:4:
../Empire/../util/Logger.h:148:47: error: 'constexpr const char* to_string(LogLevel)' redeclared as different kind of entity
  148 | constexpr const char* to_string(LogLevel level) {
      |                                               ^
../Empire/../util/Enum.h:51:18: note: previous declaration 'constexpr const std::string_view to_string'
   51 | std::string_view to_string( \
      |                  ^~~~~~~~~
../Empire/../util/Enum.h:242:5: note: in expansion of macro 'FO_DEF_ENUM_TOSTRING'
  242 |     FO_DEF_ENUM_TOSTRING(typeName, BOOST_PP_SEQ_TRANSFORM(FO_DEF_ENUM_ADD_STRING_REPR, _, values)) \
      |     ^~~~~~~~~~~~~~~~~~~~
../Empire/Diplomacy.h:12:1: note: in expansion of macro 'FO_ENUM'
   12 | FO_ENUM(
      | ^~~~~~~
../Empire/../util/Logger.h:157:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  157 |         {to_string(LogLevel::trace), LogLevel::trace},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:159:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  159 |         {to_string(LogLevel::debug), LogLevel::debug},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:161:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  161 |         {to_string(LogLevel::info),  LogLevel::info},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:163:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  163 |         {to_string(LogLevel::warn),  LogLevel::warn},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:165:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  165 |         {to_string(LogLevel::error), LogLevel::error},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:167:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  167 |         {to_string(LogLevel::min),   LogLevel::min},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~
../Empire/../util/Logger.h:169:19: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (LogLevel)'
  169 |         {to_string(LogLevel::max),   LogLevel::max},
      |          ~~~~~~~~~^~~~~~~~~~~~~~~
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::trace}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::debug}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::info}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::warn}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::error}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::min}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:171:6: error: could not convert '{<expression error>, LogLevel::max}' from '<brace-enclosed initializer list>' to 'std::pair<std::basic_string_view<char>, LogLevel>'
  171 |     }};
      |      ^
      |      |
      |      <brace-enclosed initializer list>
../Empire/../util/Logger.h:172:51: error: non-constant condition for static assertion
  172 |     static_assert(valid_names_and_levels[8].first != valid_names_and_levels[9].first);
../Empire/../util/Logger.h:173:52: error: non-constant condition for static assertion
  173 |     static_assert(valid_names_and_levels[8].second == valid_names_and_levels[9].second);
../Empire/../util/Logger.h:182:36: error: non-constant condition for static assertion
  182 | static_assert(to_LogLevel("ERROR") == LogLevel::error);
      |               ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:183:36: error: non-constant condition for static assertion
  183 | static_assert(to_LogLevel("TRACE") == to_LogLevel("trace"));
      |               ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Logger.h:184:36: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (const LogLevel&)'
  184 | static_assert(to_LogLevel(to_string(default_log_level_threshold)) == default_log_level_threshold);
      |                           ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../Empire/../util/Logger.h: In function 'std::basic_ostream<_CharT, _Traits>& operator<<(std::basic_ostream<_CharT, _Traits>&, const LogLevel&)':
../Empire/../util/Logger.h:191:20: error: no match for call to '(const std::string_view {aka const std::basic_string_view<char>}) (const LogLevel&)'
  191 |     os << to_string(level);
      |           ~~~~~~~~~^~~~~~~
../Empire/Diplomacy.cpp: In constructor 'DiplomaticMessage::DiplomaticMessage()':
../Empire/Diplomacy.cpp:24:5: error: class 'DiplomaticMessage' does not have any field named 'm_type'
   24 |     m_type(Type::INVALID)
      |     ^~~~~~
../Empire/Diplomacy.cpp:24:12: error: 'Type' has not been declared
   24 |     m_type(Type::INVALID)
      |            ^~~~
../Empire/Diplomacy.cpp: At global scope:
../Empire/Diplomacy.cpp:27:85: error: 'Type' has not been declared
   27 | DiplomaticMessage::DiplomaticMessage(int sender_empire_id, int recipient_empire_id, Type type) :
      |                                                                                     ^~~~
../Empire/Diplomacy.cpp: In constructor 'DiplomaticMessage::DiplomaticMessage(int, int, int)':
../Empire/Diplomacy.cpp:30:5: error: class 'DiplomaticMessage' does not have any field named 'm_type'
   30 |     m_type(type)
      |     ^~~~~~
../Empire/Diplomacy.cpp: In member function 'std::string DiplomaticMessage::Dump() const':
../Empire/Diplomacy.cpp:46:13: error: 'm_type' was not declared in this scope; did you mean 'wctype'?
   46 |     switch (m_type) {
      |             ^~~~~~
      |             wctype
../Empire/Diplomacy.cpp:47:10: error: 'Type' has not been declared
   47 |     case Type::WAR_DECLARATION:           retval += "War Declaration";            break;
      |          ^~~~
../Empire/Diplomacy.cpp:48:10: error: 'Type' has not been declared
   48 |     case Type::PEACE_PROPOSAL:            retval += "Peace Proposal";             break;
      |          ^~~~
../Empire/Diplomacy.cpp:49:10: error: 'Type' has not been declared
   49 |     case Type::ACCEPT_PEACE_PROPOSAL:     retval += "Accept Peace Proposal";      break;
      |          ^~~~
../Empire/Diplomacy.cpp:50:10: error: 'Type' has not been declared
   50 |     case Type::ALLIES_PROPOSAL:           retval += "Allies Proposal";            break;
      |          ^~~~
../Empire/Diplomacy.cpp:51:10: error: 'Type' has not been declared
   51 |     case Type::ACCEPT_ALLIES_PROPOSAL:    retval += "Accept Allies Proposal";     break;
      |          ^~~~
../Empire/Diplomacy.cpp:52:10: error: 'Type' has not been declared
   52 |     case Type::END_ALLIANCE_DECLARATION:  retval += "End Alliance Declaration";   break;
      |          ^~~~
../Empire/Diplomacy.cpp:53:10: error: 'Type' has not been declared
   53 |     case Type::CANCEL_PROPOSAL:           retval += "Cancel Proposal";            break;
      |          ^~~~
../Empire/Diplomacy.cpp:54:10: error: 'Type' has not been declared
   54 |     case Type::REJECT_PROPOSAL:           retval += "Reject Proposal";            break;
      |          ^~~~
../Empire/Diplomacy.cpp:55:10: error: 'Type' has not been declared
   55 |     case Type::INVALID:
      |          ^~~~
../Empire/Diplomacy.cpp: In constructor 'DiplomaticStatusUpdateInfo::DiplomaticStatusUpdateInfo()':
../Empire/Diplomacy.cpp:70:5: error: class 'DiplomaticStatusUpdateInfo' does not have any field named 'diplo_status'
   70 |     diplo_status(DiplomaticStatus::INVALID_DIPLOMATIC_STATUS)
      |     ^~~~~~~~~~~~
../Empire/Diplomacy.cpp:70:18: error: 'DiplomaticStatus' has not been declared
   70 |     diplo_status(DiplomaticStatus::INVALID_DIPLOMATIC_STATUS)
      |                  ^~~~~~~~~~~~~~~~
../Empire/Diplomacy.cpp: At global scope:
../Empire/Diplomacy.cpp:73:90: error: 'DiplomaticStatus' has not been declared
   73 | DiplomaticStatusUpdateInfo::DiplomaticStatusUpdateInfo(int empire1_id_, int empire2_id_, DiplomaticStatus status) :
      |                                                                                          ^~~~~~~~~~~~~~~~
../Empire/Diplomacy.cpp: In constructor 'DiplomaticStatusUpdateInfo::DiplomaticStatusUpdateInfo(int, int, int)':
../Empire/Diplomacy.cpp:76:5: error: class 'DiplomaticStatusUpdateInfo' does not have any field named 'diplo_status'
   76 |     diplo_status(status)
      |     ^~~~~~~~~~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage WarDeclarationDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:81:86: error: 'Type' is not a member of 'DiplomaticMessage'
   81 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::WAR_DECLARATION); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage PeaceProposalDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:84:86: error: 'Type' is not a member of 'DiplomaticMessage'
   84 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::PEACE_PROPOSAL); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage AcceptPeaceDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:87:86: error: 'Type' is not a member of 'DiplomaticMessage'
   87 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::ACCEPT_PEACE_PROPOSAL); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage AlliesProposalDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:90:86: error: 'Type' is not a member of 'DiplomaticMessage'
   90 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::ALLIES_PROPOSAL); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage AcceptAlliesDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:93:86: error: 'Type' is not a member of 'DiplomaticMessage'
   93 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::ACCEPT_ALLIES_PROPOSAL); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage EndAllianceDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:96:86: error: 'Type' is not a member of 'DiplomaticMessage'
   96 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::END_ALLIANCE_DECLARATION); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage CancelDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:99:86: error: 'Type' is not a member of 'DiplomaticMessage'
   99 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::CANCEL_PROPOSAL); }
      |                                                                                      ^~~~
../Empire/Diplomacy.cpp: In function 'DiplomaticMessage RejectProposalDiplomaticMessage(int, int)':
../Empire/Diplomacy.cpp:102:86: error: 'Type' is not a member of 'DiplomaticMessage'
  102 | { return DiplomaticMessage(sender_empire_id, recipient_empire_id, DiplomaticMessage::Type::REJECT_PROPOSAL); }
      |                                                                                      ^~~~
ninja: build stopped: subcommand failed.
```
  
</details>

--- a/Empire/Diplomacy.h
+++ b/Empire/Diplomacy.h
@@ -2,6 +2,7 @@
 #define _Diplomacy_h_
 
 #include <compare>
+#include <cstdint>
 #include <string>
 #include <boost/serialization/access.hpp>
 #include "../util/Enum.h"

--- a/universe/Enums.h
+++ b/universe/Enums.h
@@ -2,6 +2,7 @@
 #define _Enums_h_
 
 
+#include <cstdint>
 #include <iostream>
 #include <map>
 #include <string>

